src/goto-programs/remove_java_nondet.cpp

```diff
@@ -0,0 +1,330 @@
+/*******************************************************************\
+
+Module: Remove Java Nondet expressions
+
+Author: Reuben Thomas, reuben.thomas@diffblue.com
+
+\*******************************************************************/
+
+#include "goto-programs/remove_java_nondet.h"
+#include "goto-programs/goto_convert.h"
+#include "goto-programs/goto_model.h"
+
+#include "java_bytecode/java_object_factory.h"
+
+#include "util/irep_ids.h"
+
+#include <algorithm>
+#include <regex>
+
+/*******************************************************************\
```

Comment:
This change harms readability, and (counter-intuitively) *increases* the scope of the member variables. Consider:

```c++
class nondet_removert final
{
public:
  nondet_removert(...): /* set data members */ {}
  //  all the methods in the module
private:
  message_handlert &message_handler;
  symbol_tablet &symbol_table;
  size_t max_nondet_array_length;
  goto_functionst &goto_functions;
};
```

Now, all the methods in `nondet_removert` may access and modify any of the data members. Currently, each function in the module may only read the parameters that it is passed, which means that when reading the code the reader only has to keep the function parameters and local variables in their head. Making the functions into member functions means that the reader has to keep the entire state of the object in their head. Additionally, member functions would mean that each function may *modify* the data members, which creates the possibility of undesired side-effects.

This change is essentially asking for a set of unrelated variables to be shared amongst a set of unrelated functions. This is bad, in the same way that global variables are bad (albeit at a smaller scale).

Also, the conceptual need for a class is not clear. Classes serve three main purposes in C++:
- resource management via RAII
- polymorphism and customisation via virtual methods
- encapsulation/maintaining invariants over data via data hiding and public interfaces

A class would not take on any of these roles in this context, which makes me question its necessity.

Finally, this change increases the sheer quantity of code to maintain (class declaration, constructor, documentation block).